cps/oauth_bb.py

```diff
@@ -22,328 +21,183 @@
-              active=oauth_ids[0].active,
-              oauth_client_id=oauth_ids[0].oauth_client_id,
-              scope=None,
-              oauth_client_secret=oauth_ids[0].oauth_client_secret,
-              obtain_link='https://github.com/settings/developers')
-    ele2=dict(provider_name='google',
-              id=oauth_ids[1].id,
-              active=oauth_ids[1].active,
-              scope=["https://www.googleapis.com/auth/plus.me", "https://www.googleapis.com/auth/userinfo.email"],
-              oauth_client_id=oauth_ids[1].oauth_client_id,
-              oauth_client_secret=oauth_ids[1].oauth_client_secret,
-              obtain_link='https://github.com/settings/developers')
-    oauthblueprints.append(ele1)
-    oauthblueprints.append(ele2)
-
-    for element in oauthblueprints:
-        if element['provider_name'] == 'github':
-            blueprint_func = make_github_blueprint
-        else:
-            blueprint_func = make_google_blueprint
-        blueprint = blueprint_func(
-            client_id=element['oauth_client_id'],
-            client_secret=element['oauth_client_secret'],
-            redirect_to="oauth."+element['provider_name']+"_login",
-            scope = element['scope']
-        )
-        element['blueprint']=blueprint
-        app.register_blueprint(blueprint, url_prefix="/login")
-        element['blueprint'].backend = OAuthBackend(ub.OAuth, ub.session, str(element['id']),
-                                                    user=current_user, user_required=True)
-        if element['active']:
-            register_oauth_blueprint(element['id'], element['provider_name'])
-
-
-    @oauth_authorized.connect_via(oauthblueprints[0]['blueprint'])
-    def github_logged_in(blueprint, token):
-        if not token:
-            flash(_(u"Failed to log in with GitHub."), category="error")
-            return False
 
-        resp = blueprint.session.get("/user")
-        if not resp.ok:
-            flash(_(u"Failed to fetch user info from GitHub."), category="error")
-            return False
+from . import services
+
+
+oauth = None
+
+
+if services.oauth:
+    from flask import Blueprint, flash, redirect, url_for, abort, request, current_app
+    from flask_babel import gettext as _
+    from flask_login import current_user, login_user, login_required
+
+    from . import logger, ub
 
-        github_info = resp.json()
-        github_user_id = str(github_info["id"])
-        return oauth_update_token(str(oauthblueprints[0]['id']), token, github_user_id)
+    oauth = Blueprint('oauth', __name__)
+    log = logger.create()
 
 
-    @oauth_authorized.connect_via(oauthblueprints[1]['blueprint'])
-    def google_logged_in(blueprint, token):
+    # called by flask-dance on successful authorization, provides the token
+    def authorized_handler(blueprint, token):
+        provider = services.oauth.providers.get(blueprint.name)
+        assert provider, "authorized %s: no provider found" % blueprint.name
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/Ozzie-IsaacsOrganization/calibre-web/pullRequest?prid=4846306)